src/WorkflowExecuter/Services/WorkflowExecuterService.cs

```diff
@@ -319,25 +318,40 @@ private async Task<bool> UpdateWorkflowInstanceStatus(WorkflowInstance workflowI
 
         private async Task HandleDicomExport(WorkflowRevision workflow, WorkflowInstance workflowInstance, TaskExecution task, string correlationId)
         {
-            var artifactValues = GetDicomExports(workflow, workflowInstance, task);
+            var exportDestinations = workflow.Workflow?.Tasks?.FirstOrDefault(t => t.Id == task.TaskId)?.ExportDestinations;
+
+            var exportList = workflow.Workflow?.Tasks?.FirstOrDefault(t => t.Id == task.TaskId)?.ExportDestinations.Select(e => e.Name).ToArray();
+
+            var artifactValues = GetDicomExports(workflow, workflowInstance, task, exportList);
 
             if (artifactValues.Any() is false)
             {
                 await HandleTaskDestinations(workflowInstance, workflow, task, correlationId);
+
+                return;
             }
 
-            await DispatchDicomExport(workflowInstance, task, workflow.Workflow?.InformaticsGateway?.ExportDestinations, artifactValues, correlationId);
+            await DispatchDicomExport(workflowInstance, task, exportList, artifactValues, correlationId);
         }
 
-        private string[] GetDicomExports(WorkflowRevision workflow, WorkflowInstance workflowInstance, TaskExecution task)
+        private string[] GetDicomExports(WorkflowRevision workflow, WorkflowInstance workflowInstance, TaskExecution task, string[] exportDestinations)
         {
-            var exportDestinations = workflow.Workflow?.InformaticsGateway?.ExportDestinations;
+            var validExportDestinations = workflow.Workflow?.InformaticsGateway?.ExportDestinations;
 
-            if (exportDestinations is null || !exportDestinations.Any())
+            if (validExportDestinations is null || !validExportDestinations.Any()
+                || exportDestinations is null || !exportDestinations.Any())
```

Comment:
I think we have an ICollection extension for this now to do a .IsNullOrEmpty() on arrays and other collections which can simplify this.

Reply:
Arrays does not include this, but added an extension method